theia-tree-editor/src/browser/interfaces.ts

```diff
@@ -74,15 +75,15 @@ export namespace TreeEditor {
          * @param node The tree node
          * @returns The data associated with the node
          */
-        getDataForNode(node: Node): any;
+        getDataForNode(node: Node): MaybePromise<any>;
 
         /**
          * Returns the JsonSchema describing how the node's data should be rendered.
          * Alternatively, return undefined to generate a schema in the detail view.
          * @param node The tree node
          * @returns the JsonSchema describing the node's data or undefined to generate a schema.
          */
-        getSchemaForNode(node: Node): JsonSchema | undefined;
+        getSchemaForNode(node: Node): MaybePromise<JsonSchema> | undefined;
```

Comment:
I'd change the return type to `MaybePromise<JsonSchema | undefined>` because the Promise can also evaluate to `undefined`.